boa/environment.py

```diff
@@ -593,41 +204,30 @@ def deploy_code(
         gas: Optional[int] = None,
         value: int = 0,
         bytecode: bytes = b"",
-        start_pc: int = 0,
+        start_pc: int = 0,  # TODO: This isn't used
         # override the target address:
         override_address: Optional[_AddressType] = None,
     ) -> tuple[Address, bytes]:
-        if gas is None:
-            gas = self.vm.state.gas_limit
-
         sender = self._get_sender(sender)
 
-        if override_address is not None:
-            target_address = Address(override_address)
-        else:
-            nonce = self.vm.state.get_nonce(sender)
-            self.vm.state.increment_nonce(sender)
-            target_address = Address(generate_contract_address(sender, nonce))
+        target_address = (
+            self.evm.generate_contract_address(sender)
+            if override_address is None
+            else Address(override_address)
+        )
 
-        msg = Message(
-            to=constants.CREATE_CONTRACT_ADDRESS,  # i.e., b""
+        prefetch_state = self._fork_mode and self._fork_try_prefetch_state
```

Comment:
maybe fork_mode should be moved down to the evm?

Reply:
this is just a property: https://github.com/vyperlang/titanoboa/blob/3e9ac336f88989071a7e97f761c453c05ec1b670/boa/environment.py#L85

Does it make sense to move `prefetch_state` to the evm implementation?